File: frontend/src/types/event.ts
export interface Event {
  id: number;
  title: string;
  description?: string;
  date: string;
  endDate?: string;
  start_time?: string;
  startTime?: string;
  time?: string;
  end_time?: string;
  endTime?: string;
  location: string;
  type?: string;
  category?: string;
  capacity?: number;
  attendees?: number;
  maxAttendees?: number;
  registered_count?: number;
  registeredCount?: number;
  registrationDeadline?: string;
  organizer?: string;
  status?: string;
  created_at?: string;
  createdAt?: string;
}

export interface EventRegistration {
  id: string;
  eventId: string;
  memberId: string;
  registeredAt: string;
  attended: boolean;
}

File: backend/app/schemas/event.py
from pydantic import BaseModel
from typing import Optional
from datetime import date, time, datetime
from uuid import UUID
from decimal import Decimal

class EventCreate(BaseModel):
    title: str
    description: Optional[str] = None
    date: date
    start_time: time
    end_time: time
    location: str
    type: str
    capacity: Optional[int] = None
    recurring: Optional[bool] = False
    recurring_type: Optional[str] = None
    registration_deadline: Optional[date] = None
    cost: Optional[Decimal] = None
    organizer: Optional[str] = None

class EventUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    date: Optional[date] = None
    start_time: Optional[time] = None
    end_time: Optional[time] = None
    location: Optional[str] = None
    type: Optional[str] = None
    capacity: Optional[int] = None
    status: Optional[str] = None

class EventResponse(BaseModel):
    id: UUID
    title: str
    description: Optional[str]
    date: date
    start_time: time
    end_time: time
    location: str
    type: str
    capacity: Optional[int]
    registered_count: int
    status: str
    created_at: datetime
    
    class Config:
        from_attributes = True

class EventRegistrationCreate(BaseModel):
    event_id: UUID
    member_id: UUID

class EventRegistrationResponse(BaseModel):
    id: UUID
    event_id: UUID
    member_id: UUID
    registered_at: datetime
    attended: bool
    
    class Config:
        from_attributes = True
